gcsio/src/test/java/com/google/cloud/hadoop/gcsio/testing/MockGoogleCloudStorageImplFactory.java

```diff
@@ -48,4 +48,20 @@ public static GoogleCloudStorageImpl mockedGcs(HttpTransport transport) {
         storage,
         null);
   }
+
+  public static GoogleCloudStorageImpl mockedGcs(
+      GoogleCloudStorageOptions options, HttpTransport transport) {
+    Storage storage =
+        new Storage(
+            transport,
+            GsonFactory.getDefaultInstance(),
+            new TrackingHttpRequestInitializer(
+                new RetryHttpInitializer(
+                    new FakeCredentials(),
+                    RetryHttpInitializerOptions.builder()
+                        .setDefaultUserAgent("gcs-io-unit-test")
+                        .build()),
+                false));
```

Comment:
I think that this logic could be de-duplicated if you will call this new method inside the above `mockedGcs` method.